components/event/EventTabs.js

```diff
@@ -0,0 +1,174 @@
+  };
+
+  const sortUserTabItems = (tabName, order) => {
+    const { dataKey, sortKey } = getDataKeyAndSortKey(tabName);
+    sortEvents[dataKey].sort(function (a, b) {
+      console.log(sortEvents[dataKey]);
+      const aVal = sortKey.includes(".")
+        ? getNested(a, sortKey.split("."))
+        : a[sortKey];
+      const bVal = sortKey.includes(".")
+        ? getNested(b, sortKey.split("."))
+        : b[sortKey];
+      if (tabName === "My Links") {
+        if (order === "ASC") {
+          return aVal.toLowerCase() > bVal.toLowerCase()
+            ? 1
+            : aVal.toLowerCase() < bVal.toLowerCase()
+            ? -1
+            : 0;
+        } else {
+          return aVal.toLowerCase() < bVal.toLowerCase()
+            ? 1
+            : aVal.toLowerCase() > bVal.toLowerCase()
+            ? -1
+            : 0;
+        }
+      } else {
+        if (order === "ASC") {
+          return new Date(aVal) > new Date(bVal)
+            ? 1
+            : new Date(aVal) < new Date(bVal)
+            ? -1
+            : 0;
+        } else {
+          return new Date(aVal) < new Date(bVal)
+            ? 1
+            : new Date(aVal) > new Date(bVal)
+            ? -1
+            : 0;
+        }
+      }
+    });
+    setSortEvents({ ...sortEvents });
+  };
+
+  const getNested = (obj, args) => {
+    return args.reduce((obj, level) => obj && obj[level], obj);
+  };
+
+  return (
+    <div>
+      <div className="sm:hidden">
+        <label htmlFor="tabs" className="sr-only">
+          Select a tab
+        </label>
+        <select
+          id="tabs"
+          name="tabs"
+          onChange={changeTab}
+          className="block w-full rounded-md border-gray-300 focus:border-indigo-500 focus:ring-indigo-500"
+          defaultValue={tabs.find((tab) => tab.current)?.title}
+        >
+          {tabs.map((tab) => (
+            <option key={tab.key}>{tab.title}</option>
+          ))}
+        </select>
+      </div>
+      <div className="hidden sm:block">
+        <div className="border-b border-gray-200">
+          <nav className="-mb-px flex" aria-label="Tabs">
+            {tabs.map((tab) => (
+              <a
+                key={tab.key}
+                href={tab.href}
+                onClick={(e) => changeTab(e, tab.title)}
+                className={classNames(
+                  tab.current
+                    ? "border-indigo-500 text-indigo-600"
+                    : "border-transparent text-gray-500 hover:text-gray-700 hover:border-gray-300",
+                  "w-1/4 py-4 px-1 text-center border-b-2 font-medium text-sm flex justify-center items-center gap-2 cursor-pointer"
+                )}
+                aria-current={tab.current ? "page" : undefined}
+              >
+                {tab.key === "all" && (
+                  <FaListUl title="Show All" className="text-base" />
+                )}
+                {tab.key === "virtual" && (
+                  <MdOutlineOnlinePrediction
```

Comment:
Oh yeah, fixed this
![image](https://user-images.githubusercontent.com/95094057/211855963-4a3742e4-049e-4db6-bd34-0ab02ec55569.png)

Reply:
> @ChinmayMhatre hey, I'm having an issue with the sorting function, what happens is the sort works on the second click it doesn't work on the first click but then it starts working after the first click can you have a look at that functions code whenever you are free, thanks

I went Through the issue and noticed the issue was majorly discussing creating tabs instead of the four button.
I think we could focus on refactoring the tabs further and create a new issue for the sorting. You could take on that issue if you want since you had been working on this.